packages/search/src/components/InputSearch/index.tsx

```diff
@@ -0,0 +1,70 @@
+import React, { useLayoutEffect, useRef, useState } from 'react';
+import { MagnifyingGlass } from '@rainbow-modules/icons';
+import { InputComponent, InputSearchProps } from './types';
+import { StyledContainer, StyledInput } from './styled';
+import Trailing from './trailing';
+
+const InputSearch: React.FC<InputSearchProps> = ({
+    className,
+    style,
+    disabled,
+    readOnly,
+    name,
+    value,
+    variant,
+    onChange,
+    onSearch,
+}: InputSearchProps) => {
+    const trailingRef = useRef<HTMLDivElement>(null);
+    const inputRef = useRef<InputComponent>(null);
+    const [paddingRight, setPaddingRight] = useState<number>();
+
+    useLayoutEffect(() => {
+        if (trailingRef.current) {
+            const { width } = trailingRef.current.getBoundingClientRect();
+            if (paddingRight !== width) setPaddingRight(width);
+        }
+    }, [paddingRight, value]);
+
+    const handleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+        if (onChange) onChange(event.target.value);
+    };
+
+    const handleKeyDown = (event: React.KeyboardEvent<HTMLInputElement>) => {
+        if (onSearch && event.key === 'Enter') onSearch(value);
+    };
+
+    const clear = () => {
+        if (onChange) onChange('');
+        if (inputRef.current) inputRef.current.focus();
+    };
+
+    return (
+        <StyledContainer className={className} style={style}>
+            <StyledInput
+                name={name}
+                icon={<MagnifyingGlass />}
+                paddingRight={paddingRight}
+                value={value}
+                disabled={disabled}
+                readOnly={readOnly}
+                onChange={handleChange}
+                onKeyDown={handleKeyDown}
+                ref={inputRef}
```

Comment:
to have all the functionality of a native search input we should pass it `type ="search"`